docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+```kotlin title="build.gradle"
+buildscript {
+  ext.safeExtGet = {prop, fallback ->
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+  }
+  repositories {
+    google()
+    gradlePluginPortal()
+  }
+  dependencies {
+    classpath("com.android.tools.build:gradle:7.1.1")
+  }
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'com.facebook.react'
+
+android {
+  compileSdkVersion safeExtGet('compileSdkVersion', 31)
+
+  defaultConfig {
+    minSdkVersion safeExtGet('minSdkVersion', 21)
+    targetSdkVersion safeExtGet('targetSdkVersion', 31)
+    buildConfigField("boolean", "IS_NEW_ARCHITECTURE_ENABLED", "true")
+  }
+}
+
+repositories {
+  maven {
+    // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+    url "$projectDir/../node_modules/react-native/android"
+  }
+  mavenCentral()
+  google()
+}
+
+dependencies {
+  implementation 'com.facebook.react:react-native:+'
+}
+
+react {
+    jsRootDir = file("../js/")
+    libraryName = "RTNCalculator"
+    codegenJavaPackageName = "com.RTNCalculator"
+}
+```
+
+Of interest in the `build.gradle` file:
+
+- The `defaultConfig` block, within the `android` block, adds a `buildConfigField` to enable the New Architecture.
+- The `react` block configures the CodeGen process. For Android, we need to specify:
+  - the `jsRootDir`, which contains the relative path to the JavaScript specs
+  - the `libraryName` we will use to link the library in the app.
+  - the `codegenJavaPackageName` which corresponds to the name of the Java package we will use for the code generated by **CodeGen**.
+
+## 4. Native Code
+
+For the final step in getting your TurboModule ready to go, you'll need to write some native code to connect the JavaScript side to the native platforms. This process requires two main steps:
+
+- Run **CodeGen** to see what it generates.
+- Write your native code, implementing the generated interfaces.
+
+When developing a React Native app that uses a TurboMOdule, it is responsibility of the app to actually generate the code using **CodeGen**. However, when developing a TurboModule as a library, we need to reference the generated code, and it is therefore useful to see what the app will generate.
+
+As first step for both iOS and Android, this guide shows how to execute manually the scripts used by **CodeGen** to generate the required code. Further information on **CodeGen** can be found [here](/docs/pillars-codegen.md)
+
+:::caution
+The code generated by the **CodeGen** in this step should not be committed to the versioning system. React Native apps are able to generate the code when the app is built. This allows to avoid any ABI incompatibility and to ensure that a consistent version of the **CodeGen** is used.
+:::
+
+### iOS
+
+#### Generate the code - iOS
+
+To run Codegen for the iOS platform, we need to open a terminal and run the following command:
+
+```sh title="Running CodeGen for iOS"
+cd MyApp
+yarn add ../RTNCalculator
+cd ..
```

Comment:
> the extra ..

By this, do you mean in the `yarn add` command? That would be needed, yarn would need to be run from `MyApp`

As for the rest, I copied the way the Fabric guide did it.

Reply:
We can remove the `cd ..` line, but in that case we have to update the next command as it follows:
```sh
node node_modules/react-native/scripts/generate-artifacts.js \
  --path . \
  --outputPath ../RTNCalculator/generated/
```
I think that the command is clearer if we run it from the parent folder, using `cd ..`, because it uses less `.` and `..` which adds more cognitive load for the reader.. What do you think?

Reply:
I also think it's a bit clearer from the parent. I'd rather be a bit more verbose for clarity's sake.